mlflow/pipelines/pipeline.py

```diff
@@ -153,15 +153,16 @@ def _get_step(self, step_name) -> BaseStep:
         return self._steps[step_names.index(step_name)]
 
     @experimental
-    @abc.abstractmethod
     def _get_subgraph_for_target_step(self, target_step: BaseStep) -> List[BaseStep]:
         """
         Return a list of step objects representing a connected DAG containing the target_step.
         The returned list should be a sublist of self._steps.
-
-        Concrete pipeline class should implement this method.
         """
-        pass
+        subgraph = []
+        for step in self._steps:
+            if target_step.is_predict_step() == step.is_predict_step():
+                subgraph.append(step)
```

Comment:
This would mean that in the all future pipelines, we'll have 2 graphs at least: training graph and inference graph. It seems fine to me but can you add a comment to state this explicitly? It'd be helpful for future readers and new pipeline implementers.

Reply:
Not sure that making this assumption is worthwhile, as it'll likely break in the future. Is including the subgraph logic in each pipeline implementation too cumbersome?

Reply:
I can update this to an enum, though I don't see any reason to include the subgraph logic in each pipeline implementation - not that it's terribly cumbersome, just unnecessarily not DRY. Do you see things differently?

"This would mean that in the all future pipelines, we'll have 2 graphs at least: training graph and inference graph." - if a pipeline consists only of training steps, it would have a single subgraph.